### ISSUE TITLE
[MCHECKSTYLE-31] Multi-module reports do not support custom classpath configurations

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-checkstyle-plugin#186]
**[Mike Perham](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mperham)** opened **[MCHECKSTYLE-31](https://issues.apache.org/jira/browse/MCHECKSTYLE-31?redirect=false)** and commented

The latest multi-module tip shows the following:

```
<reporting>
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-checkstyle-plugin</artifactId>
        <configuration>
          <configLocation>whizbang/checkstyle.xml</configLocation>
          <headerLocation>whizbang/LICENSE.txt</headerLocation>
        </configuration>
        <dependencies>
          <dependency>
            <groupId>com.example.whizbang</groupId>
            <artifactId>build-tools</artifactId>
            <version>1.0</version>
          </dependency>
        </dependencies>
      </plugin>
    </plugins>
  </reporting>
```

This is invalid according to the latest 2.0.2 POM schema.  \<dependencies> is not supported in reporting plugins.

So it seems impossible to provide a custom config in a multi-module build without using a network URL as we cannot use File or classpath.

---

**Affects:** 2.0-beta-1

**Issue Links:**
- [MSITE-444](https://issues.apache.org/jira/browse/MSITE-444) reporting section should allow additional dependencies in plugins as well
 (_**"depends upon"**_)

3 votes, 7 watchers

**Comment by jira-importer** [comment]
**[skaze](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=skaze)** commented

Until core gets reporting sorted (ie. reportingManagement, dependencies for reports, etc) we define the checkstyle, clover-2.0 and pmd plugins as \<build> plugins and specify their dependencies in the nornmal plugin way. Fortunately they dont have default goals bound to the lifecycle so don't do any harm in the normal biuld but when the report or check mojos are invoked either by site or via explicit invocation the plugin configuration seems to inherit the \<build> defined configuration, including its dependencies.

**Comment by jira-importer** [comment]
**[Guillaume Boué](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gboue)** commented

This is fixed since the Site plugin 3.0 (MSITE-444). You can declare dependencies in `pluginManagement` section and it will be inherited by report plugins like checkstyle.
